help/Readme.md

```diff
@@ -0,0 +1,114 @@
+The Packer process initializes a connection to Azure subscription via Azure CLI, and automatically creates the temporary Azure resources required to build the source VM(temporary resource group, network interfaces, and VM from the "clean" image specified in the template). 
+If deployment succeeded, the build agent connects to the deployed VM and starts to execute installation steps from the selected template.
+In case if any template step fails, image generation will be aborted and temporary VM will be terminated.
+After successfull image generation, snapshot of temporary VM will be converted to VHD image and then uploaded to the specified Azure Storage Account.
+
+### Build Agent requirements
+- `OS` - Windows/Linux
+- `packer` - Can be downloaded from https://www.packer.io/downloads
+- `PowerShell 5.0 or higher` or `PSCore` for linux distributes.
+- `Azure CLI ` - https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest
+
+#### Azure DevOps self-hosted pool requirements
+To connect to a temporary VM packer use WinRM or SSH connections on public IP interfaces.
+If you use a build agent located in an Azure subscription, please make sure that HTTPS/SSH ports are allowed for incoming/outgoing connections.
+In case of firewall restrictions, prohibiting connections from public addresses, private virtual network resources can be deployed and passed as arguments to the packer. This approach allows virtual machines to use private connections inside VLAN.
+
+#### Service principal
+Packer uses Service Principal to authorize in Azure infrastructure. To setup image-generation CI or use packer manually — SP with full read-write permissions for selected Azure subscription needed.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+You can use local machine as build agent, or deploy virtual machine in [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli).
+
+Download `packer` from https://www.packer.io/downloads, or install it via Chocolately.
+```
+Choco-Install -PackageName packer
+```
+
+Install Azure CLI - https://docs.microsoft.com/ru-ru/cli/azure/install-azure-cli-windows?view=azure-cli-latest&tabs=azure-cli.
+```
+Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm .\AzureCLI.msi
+```
+
+Download Virtual-Environments repository.
+```
+git clone https://github.com/actions/virtual-environments.git
+```
+
+Import [GenerateResourcesAndImage](https://github.com/actions/virtual-environments/blob/main/helpers/GenerateResourcesAndImage.ps1) script from `/helpers` folder, and run `GenerateResourcesAndImage` function in Powershell.
```

Comment:
Looks like we need to add Ubuntu 20 to that script

Reply:
Right. Fixed.